### PR TITLE
libkb: serialize keys in right order to preserve signatures on subkeys     

### PR DESCRIPTION
**Pull request by maxtaco**
- solves the 'ERROR key generation error: Failed to get pubkey: openpgp: invalid data: subkey signature invalid: openpgp: invalid data: signing
- change revoke protocol to take a KID, rather than a string                                       
- resurrect Test454
- put tests into place; turns out this fix broke another test!                                                                                 
- fix regression introduced with previous tests

### CODE REVIEW

**Comment by maxtaco** [comment]
@patrickxb PTAL?

File: protocol/avdl/revoke.avdl [hunk 1]
```diff
@@ -3,7 +3,7 @@
 protocol revoke {
   import idl "common.avdl";
 
-  void revokeKey(int sessionID, string keyID);
+  void revokeKey(int sessionID, KID keyID);
```
Comment (maxtaco) [inline protocol/avdl/revoke.avdl]: this came along for the ride, but figured i'd fix it

File: go/libkb/skb.go [hunk 2]
```diff
@@ -93,25 +93,44 @@ func (key *PGPKeyBundle) ToLksSKB(lks *LKSec) (ret *SKB, err error) {
 		return nil, fmt.Errorf("nil lks")
 	}
 	var pk, sk bytes.Buffer
+	ret = NewSKB(lks.G())
+
+	serializePublic := func() error { return key.Entity.Serialize(&pk) }
```
Comment (maxtaco) [inline go/libkb/skb.go]: this is the crux of the fix.

File: go/libkb/pgp_key.go [hunk 3]
```diff
@@ -32,12 +32,19 @@ type PGPKeyBundle struct {
...
+
+	// True if this key was generated by this program
+	Generated bool
 }
 
 func NewPGPKeyBundle(entity *openpgp.Entity) *PGPKeyBundle {
 	return &PGPKeyBundle{Entity: entity}
 }
 
+func NewGeneratedPGPKeyBundle(entity *openpgp.Entity) *PGPKeyBundle {
```
Comment (maxtaco) [inline go/libkb/pgp_key.go]: need to distinguish the case of a generated key from an imported key

File: go/engine/pgp_import_key_test.go [hunk 4]
```diff
@@ -363,3 +371,97 @@ ejqLyhhGTnPydDtFrzsGkw3Qz5r1K2p88A2mGkix3/H0CnkcSSTxI/ID2OQndQOZ
 LpRtxqDTDVA6H/R+dqEhg/ni2jAapEr4VzIbew==
 =bXrj
 -----END PGP PRIVATE KEY BLOCK-----`
+
+const keyIssue2147 = `-----BEGIN PGP PRIVATE KEY BLOCK-----
```
Comment (maxtaco) [inline go/engine/pgp_import_key_test.go]: throw-away key that repro's the issue

**Comment by patrickxb** [comment]
wow, nice.

LGTM.

So no more pgp bugs? :)

**Comment by maxtaco** [comment]
not so far as i know... but you never know...

**Pull request merged**